domain/miningmanager/mempool/mempool_utxo_set.go

```diff
@@ -0,0 +1,76 @@
+package mempool
+
+import (
+	"fmt"
+
+	"github.com/kaspanet/kaspad/domain/consensus/utils/consensushashing"
+
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/utxo"
+	"github.com/kaspanet/kaspad/domain/miningmanager/mempool/model"
+)
+
+type mempoolUTXOSet struct {
+	mempool                       *mempool
+	poolUnspentOutputs            model.OutpointToUTXOEntry
+	transactionByPreviousOutpoint model.OutpointToTransaction
+}
+
+func newMempoolUTXOSet(mp *mempool) *mempoolUTXOSet {
+	return &mempoolUTXOSet{
+		mempool:                       mp,
+		poolUnspentOutputs:            model.OutpointToUTXOEntry{},
+		transactionByPreviousOutpoint: model.OutpointToTransaction{},
+	}
+}
+
+// this function MUST be called with the mempool mutex locked for writes
+func (mpus *mempoolUTXOSet) addTransaction(transaction *model.MempoolTransaction) {
+	outpoint := &externalapi.DomainOutpoint{TransactionID: *transaction.TransactionID()}
+
+	for i, input := range transaction.Transaction().Inputs {
+		outpoint.Index = uint32(i)
+
+		delete(mpus.poolUnspentOutputs, *outpoint)
```

Comment:
I think we want to check whether it existed in poolUnspentOutputs in the first place as well.

Reply:
Why?

If it didn't exist this means the output was created in consensus, which is perfectly fine.